chat-server/server.js

```diff
@@ -25,6 +26,31 @@ app.get("/", (request, response) => {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  messages.push(newMessage);
+  res.send("Message added");
+});
+
+app.get("/messages", (req, res) => {
+  res.send(messages);
+});
+
+app.get("/messages/:id", (req, res) => {
+  const messageId = parseInt(req.params.id);
+  const message = messages.find((obj) =>  obj.id === messageId );
+  if (!message) return res.status(404).json({ message: "Message does not exist" });
+  res.json(message);
+});
+
+app.delete("/messages/:id", (req, res) => {
+  const index = messages.findIndex((obj) => obj.id === parseInt(req.params.id));
+  if (index === -1) return res.status(404).json({ message: "No message to delete" });
```

Comment:
Even though your code here is correct and works, a lot of programmers will always add `{}`s around if bodies, even `if` they're just one statement - have a read of https://www.synopsys.com/blogs/software-security/understanding-apple-goto-fail-vulnerability-2.html for some explanation as to why :)